### PR TITLE
Core/Spells: Generic berserk spell taunt immunity

### PR DESCRIPTION
**Pull request by Krudor**
**Changes proposed**: Add taunt immunity to units with the generic berserk spell (Id: 26662), as the spell description states.

**Target branch(es)**: 6x

### CODE REVIEW

**Comment by Krudor** [comment]
I need to learn to squash my base branch at some point -_-

**Comment by Aokromes** [comment]
No need anymore.

**Comment by Shauren** [comment]
Not sure if this is correct, after all taunt diminishing or complete immunity for berserk was added exactly because of some top guild playing ping-pong with Baleroc when he enraged! Definitely not for 3.3.5

**Comment by Krudor** [comment]
Modified main post to specify 6.x only

**Comment by Ulduar** [comment]
Baleroc 25 HM in actual time. Hunters kite boss in enrage, boss had a health of 217 million, is also only 6 minutes berserk. Finished boss in the Berserker Rage using hunters and kite the boss.

File: src/server/scripts/Spells/spell_generic.cpp [hunk 1]
```diff
@@ -4386,6 +4386,41 @@ class spell_gen_azgalor_rain_of_fire_hellfire_citadel : public SpellScriptLoader
...
+            void OnApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+            {
+                GetTarget()->ApplySpellImmune(0, IMMUNITY_STATE, SPELL_AURA_MOD_TAUNT, true);
+                GetTarget()->ApplySpellImmune(0, IMMUNITY_EFFECT, SPELL_EFFECT_ATTACK_ME, true);
+            }
+
+            void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+            {
+                GetTarget()->ApplySpellImmune(0, IMMUNITY_STATE, SPELL_AURA_MOD_TAUNT, false);
+                GetTarget()->ApplySpellImmune(0, IMMUNITY_EFFECT, SPELL_EFFECT_ATTACK_ME, false);
```
Comment (Meruemu) [inline src/server/scripts/Spells/spell_generic.cpp]: What if the creature has another aura which grants taunt immunity? Might these lines cause some issues?

Comment (Krudor) [inline src/server/scripts/Spells/spell_generic.cpp]: What would you do to circumvent this?

Comment (Meruemu) [inline src/server/scripts/Spells/spell_generic.cpp]: Sadly, i have no clue for now :confused: I just wanted to point a possible problem.

**Comment by Aokromes** [comment]
what's missing here? this can be merged? @Shauren @joschiwald

**Comment by Treeston** [comment]
Baleroc was specifically tauntable, it uses a different berserk aura. That was intentional.

Other bosses were untauntable even in 3.3.5 (check 3.3.5 description to make sure?)

**Comment by Aokromes** [comment]
This branch has conflicts that must be resolved
 Use the command line to resolve conflicts before continuing.
Conflicting files
src/server/scripts/Spells/spell_generic.cpp

**Comment by Shauren** [comment]
This is already handled by core
https://github.com/TrinityCore/TrinityCore/blob/a16d3209fd1cbc39b9b9fd5219b85419ed9d775c/src/server/game/Spells/SpellInfo.cpp#L3108

**Pull request closed without merging**